common-library/src/main/java/com/yas/commonlibrary/exception/ApiExceptionHandler.java

```diff
@@ -85,6 +86,27 @@ protected ResponseEntity<ErrorVm> handleInternalServerErrorException(InternalSer
         return ResponseEntity.internalServerError().body(errorVm);
     }
 
+    @ExceptionHandler(MissingServletRequestParameterException.class)
+    protected ResponseEntity<ErrorVm> handleMissingParams(MissingServletRequestParameterException e) {
+        return handleBadRequest(e, null);
+    }
+
+    @ExceptionHandler(ResourceExistedException.class)
+    public ResponseEntity<ErrorVm> handleResourceExistedException(ResourceExistedException ex, WebRequest request) {
+        HttpStatus status = HttpStatus.CONFLICT;
+        String message = ex.getMessage();
+
+        return buildErrorResponse(status, message, null, ex, request, 409);
```

Comment:
Hi, since you're already creating a new HttpStatus object, you can use its `value()` method to retrieve the HTTP code value instead of using a number like 409.

Reply:
The hard-coded sections come from previous code. I’ve moved all the exception handlers to a common library for reuse and to shorten the code. Your idea is great, but I think we should consider it in another PR with a different issue, as some handler methods set the status code to 0 with HttpStatus.BAD_REQUEST (which is usually 400). We need to take some time to understand the author’s intent with this code.

```
    @ExceptionHandler({ConstraintViolationException.class})
    public ResponseEntity<ErrorVm> handleConstraintViolation(ConstraintViolationException ex) {
        HttpStatus status = HttpStatus.BAD_REQUEST;

        List<String> errors = ex.getConstraintViolations().stream()
            .map(violation -> String.format("%s %s: %s",
                violation.getRootBeanClass().getName(),
                violation.getPropertyPath(),
                violation.getMessage()))
            .toList();

        return buildErrorResponse(status, "Request information is not valid", errors, ex, null, 0);
    }
```

Reply:
Agreed!